epaint/src/textures.rs

```diff
@@ -27,18 +41,29 @@ impl TextureManager {
     /// MUST have a white pixel at (0,0) ([`crate::WHITE_UV`]).
     ///
     /// The texture is given a retain-count of `1`, requiring one call to [`Self::free`] to free it.
-    pub fn alloc(&mut self, name: String, image: ImageData) -> TextureId {
+    pub fn alloc(&mut self, name: String, image: ImageData, filter: TextureFilter) -> TextureId {
         let id = TextureId::Managed(self.next_id);
         self.next_id += 1;
 
-        self.metas.entry(id).or_insert_with(|| TextureMeta {
-            name,
-            size: image.size(),
-            bytes_per_pixel: image.bytes_per_pixel(),
-            retain_count: 1,
-        });
-
-        self.delta.set.insert(id, ImageDelta::full(image));
+        let filter = self
+            .metas
+            .entry(id)
+            .or_insert_with(|| TextureMeta {
+                name,
+                filter,
+                size: image.size(),
+                bytes_per_pixel: image.bytes_per_pixel(),
+                retain_count: 1,
+            })
+            .filter;
+
+        self.delta.set.insert(
+            id,
+            TextureDelta {
+                filter,
+                image: ImageDelta::full(image),
```

Comment:
did you consider making `filter` part of `ImageDelta` instead of adding a new `TextureDelta`? The new code is quite onion:y, and I don't think `ImageDelta` is ever used without being part of a `TextureDelta` anyway?